contribute/design/1.x/csi/20190107-cStor-volume-resize.md

```diff
@@ -0,0 +1,357 @@
+    check is there any capacity difference in CV and CVC if there is no change
+    then update CVC status condition as resize success and release the lease.
+    If capacity change is noticed after taking lease, repeat this step after
+    releasing lease(left as is reconciliation will do rework).
+
+    Example status: CVC status when resize is success
+```yaml
+      Status:
+        capacity: 10Gi
+        Conditions:
+          - Type: Resizing
+            Status: Success
+            LastProbeTime: date
+            LastTransitionTime: (update while updating resize condition status)
+            Reason: Capacity changed
+            Message: Resize Success
+```
+   Note: Please refer CVC under Custom Resources used to resize cStor volume section
+         for entier CVC.
+
+- Cstor-volume-mgmt container as a target side car which is already having a
+  volume-controller watching on CV object will get update event (if event is missed
+  volume controller will process during sync event) then volume-controller will process
+  the request with following steps
+    - Volume-controller will executes `istgtcontrol resize <size_unit>`
+      command(rpc call) if there is any resize status is pending on that
+      volume(based on resize status condition).
+    - If above rpc call success volume-controller will updates istgt.conf file
+      alone but not CVR capacity[why? CVR capacity is treated initial volume
+      capacity and maintaining initial capacity will be helpful in ephemeral
+      case]. If volume controller succeed in updating istgt.conf file then update
+      status capacity and success status for resize condition on CV.
+    - If it is a failure response from rpc then generate kubernetes events and
+      do nothing just return(reconciliation will take care of doing the above
+      process again).
+- When cstor-istgt get a resize request it will trigger a resize request to replica
+  and return a success response to cstor-volume-mgmt[why? cstor-istgt will not concern
+  whether resize request is success or failure. If resize request succeed then
+  there is no problem. If resize request fails and if replica gets IO greater
+  than it's size then IO will be failed on that replica and cstor-istgt will
+  disconnect IO failure replica. As part of replica's data connection, target
+  will exchange the size information with replica. During data connection replica
+  checks if it is single replica volume then it will resize main volume else it
+  will resize the internal rebuild clone volume].
+  Note: Processing `istgtcontrol resize` is a blocking call.
+
+- zfs will receives the resize request, it will resize the corresponding volume
+  and sent back the response to cstor-istgt(aka target).
+
+Once resize operation is succeed at OpenEBS side CSI node plugin(/kubelet) will
+trigger resize operation on filesystem level as part of online resizing.
+
+## High Level cStor Volume Resize Workflow
+  ![Resize Workflow](Resize_work_flow.png)
```

Comment:
add relative path like - [./Resize_work_flow.png]

Reply:
Done!! Thanks for review comments @kmova .